src/Plugins/DBFTPlugin/Consensus/ConsensusService.cs

```diff
@@ -85,29 +90,30 @@ private void InitializeConsensus(byte viewNumber)
                 Log($"View changed: view={viewNumber} primary={context.Validators[context.GetPrimaryIndex((byte)(viewNumber - 1u))]}", LogLevel.Warning);
             Log($"Initialize: height={context.Block.Index} view={viewNumber} index={context.MyIndex} role={(context.IsPrimary ? "Primary" : context.WatchOnly ? "WatchOnly" : "Backup")}");
             if (context.WatchOnly) return;
+            blockGenTime = neoSystem.GetBlockGenTime();
```

Comment:
You need to initialize it only at view 0, because block time can't be updated between views.

Reply:
> You need to initialize it only at view 0, because block time can't be updated between views.

This does not matter, pelase focus on logic, optimization can be done at any time. As long as the logic is right, should be fine.

Reply:
It's not an optimisation, it's the logic issue, and it needs to be fixed. The node may get some new block from the network, and it's just not correct to retrieve the updated block generation time if the node is trying to change its view at this moment. In this situation new block generation time should be applied only starting from the next height, when consensus is initialized at view 0.

Reply:
Would you mind to have a brand new pr to replace mine, i think i am lost here and is not able to follow your directions. Would definatly be much easier and faster if you could do it directly.

Reply:
Just move `blockGenTime = neoSystem.GetBlockGenTime()` to `context.Reset` function under `if (viewNumber == 0)` condition, that's it.

Reply:
Implemented in [f167925](https://github.com/neo-project/neo/pull/3622/commits/f1679256226ee98a7e2476b7ec3bab23b3693270).